### PR TITLE
 feat: implement codemod ibm-products-update-userprofileimage

### LINKED ISSUE #6308: codemod to transform from `UserProfileImage` to `UserAvatar`
Write a Codemod to transform `UserProfileImage` to `UserAvatar`.

- [x] Implement the codemod
- [x] Include fixtures in the `packages/upgrade/transforms/__testfixtures__` and tests in the `packages/upgrade/transforms/__tests__` directory.
- [x] Include in the `list` command by updating the `packages/upgrade/src/upgrades.js` file.

### PR DESCRIPTION
**Pull request by anamikaanu96**
Closes [6308](https://github.com/carbon-design-system/ibm-products/issues/6308)

This PR updates the @carbon/upgrade package with a new transform ibm-products-update-userprofileimage, to update the  UserProfileImage to UserAvatar.

#### Changelog

**New**

- New transform packages/upgrade/transforms/ibm-products-update-userprofileimage.js
- Include fixtures in the packages/upgrade/transforms/__testfixtures__/ibm-products-update-userprofileimage.input.js and packages/upgrade/transforms/__testfixtures__/ibm-products-update-userprofileimage.output.js files
- Include tests packages/upgrade/transforms/__tests__/ibm-products-update-userprofileimage-test.js
- Implement a sample packages/upgrade/fixtures/ibm-products-update-userprofileimage-sample.jsx to test the Codemod locally

**Changed**

- Update packages/upgrade/src/upgrades.js with a new migration entry ibm-products-update-userprofileimage

#### Testing / Reviewing

Try out the Codemod locally by:
- Pulling down this PR
- Run cd packages/upgrade
- Run yarn build to build the CLI
- Run `./bin/carbon-upgrade.js migrate list` to verify a new entry ibm-products-update-userprofileimage (Rewrites UserProfileImage to UserAvatar) is present
- Run `npx jest transforms/__tests__/ibm-products-update-userprofileimage-test.js` and verify test result
- Run `./bin/carbon-upgrade.js migrate ibm-products-update-userprofileimage fixtures/ibm-products-update-userprofileimage-sample.jsx -w` to try the Codemod. ( Select the /carbon/packages/upgrade as workspace)

### CODE REVIEW

File: packages/upgrade/fixtures/ibm-products-update-userprofileimage-sample.jsx
Comment (tay1orjones) [inline packages/upgrade/fixtures/ibm-products-update-userprofileimage-sample.jsx:1]: Is this file necessary? How is it used?

**Review by tay1orjones — COMMENTED** [review]
Looks good, just one question

File: packages/upgrade/fixtures/ibm-products-update-userprofileimage-sample.jsx
Comment (anamikaanu96) [reply packages/upgrade/fixtures/ibm-products-update-userprofileimage-sample.jsx:1]: Hey @tay1orjones Thanks for reviewing! 
This file is a sample file which is used for demo purpose.

Comment (tay1orjones) [reply packages/upgrade/fixtures/ibm-products-update-userprofileimage-sample.jsx:1]: Okay - that's typically what the files in __testfixtures__ are for, and I see a lot of overlap there. I think we could remove this file.

**Review by tay1orjones — APPROVED** [review]

**Review by ariellalgilmore — APPROVED** [review]
LGTM!
